frontends/infinite-corridor/src/factories/search.ts

```diff
@@ -0,0 +1,148 @@
+    throw new Error("unknown type")
+  }
+
+  return {
+    _id:     `id_String${casual.random}`,
+    _source: hit
+  }
+}
+
+export const makeCourseResult = (): LearningResourceResult => ({
+  id:                casual.integer(1, 1000),
+  title:             casual.title,
+  url:               casual.url,
+  image_src:         "http://image.medium.url",
+  short_description: casual.description,
+  platform:          casual.random_element(["edx", "ocw"]),
+  offered_by:        [casual.random_element(["edx", "ocw"])],
+  topics:            [casual.word, casual.word],
+  object_type:       "course",
+  runs:              R.times(makeRun, 3),
+  is_favorite:       casual.coin_flip,
+  lists:             casual.random_element([[], [100, 200]]),
+  audience:          casual.random_element([
+    [],
+    [OPEN_CONTENT],
+    [PROFESSIONAL],
+    [OPEN_CONTENT, PROFESSIONAL]
+  ]),
+  certification: casual.random_element([[], [CERTIFICATE]])
+})
+
+export const makeProgramResult = (): LearningResourceResult => ({
+  id:                casual.integer(1, 1000),
+  title:             casual.title,
+  url:               casual.url,
+  image_src:         "http://image.medium.url",
+  short_description: casual.description,
+  topics:            [casual.word, casual.word],
+  object_type:       "program",
+  offered_by:        [casual.random_element(["xpro", "micromasters"])],
+  runs:              [makeRun()],
+  is_favorite:       casual.coin_flip,
+  lists:             casual.random_element([[], [100, 200]]),
+  audience:          casual.random_element([
+    [],
+    [OPEN_CONTENT],
+    [PROFESSIONAL],
+    [OPEN_CONTENT, PROFESSIONAL]
+  ]),
+  certification: casual.random_element([[], [CERTIFICATE]])
+})
+
+export const makeRun = (): LearningResourceRun => {
+  return {
+    id:               casual.integer(1, 1000),
+    url:              casual.url,
+    language:         casual.random_element(["en-US", "fr", null]),
+    semester:         casual.random_element(["Fall", "Spring", null]),
+    year:             casual.year,
+    level:            casual.random_element(["Graduate", "Undergraduate", null]),
+    start_date:       casual.date(DATE_FORMAT),
+    end_date:         casual.date(DATE_FORMAT),
+    best_start_date:  casual.date(DATE_FORMAT),
+    best_end_date:    casual.date(DATE_FORMAT),
+    enrollment_start: casual.date(DATE_FORMAT),
+    enrollment_end:   casual.date(DATE_FORMAT),
+    availability:     casual.random_element(["archived", "current", "Upcoming"]),
+    instructors:      [
+      {
+        first_name: casual.name,
+        last_name:  casual.name,
+        full_name:  casual.name
+      },
+      {
+        first_name: casual.name,
+        last_name:  casual.name,
+        full_name:  casual.name
+      }
+    ],
+    prices: [{ mode: "audit", price: casual.integer(1, 1000) }]
+  }
+}
+
+export const makeSearchResponse = (
+  //@ts-expect-error SETTINGS throws an error
+  pageSize: number = SETTINGS.search_page_size,
```

Comment:
**Requested Change**

In `frontends/infinite-corridor/src/types/global.d.ts`, could you add
```ts
/* eslint-disable no-var */

/**
 * Settings injected by Django
 */
interface SETTINGS {
  search_page_size: number;
}
export declare global {
  declare var SETTINGS: SETTINGS
}
```

Then TS will know SETTINGS is globally available.

IMO we should make settings globally available in infinite-corridor, but not in say, `ol-search-ui`. We can pass the embedly key and ocw base url as props to LEarningResourceCard. But I don't have a strong opinion about this.